### DISCUSSION TITLE
如何实现父路径 /home 渲染组件A，子路径 /home/b 渲染组件B

### DISCUSSION THREAD

**Discussion by xliez** [umijs/umi#11832] [Q&A] [answered]
需求：比如/home 是渲染组件A，/home/b 是渲染组件B

目前按如下配置的话，在子路径下依然是渲染组件A
```
{
  path: '/home', component: A, 
  routes: [{ 
     path: '/home/b', component: B
  }]
}
```

umi 版本： "@umijs/max": "^4.0.87"

**Comment by fz6m** [comment, marked as answer]
umi 4 使用的是 react router v6 ，嵌套路由需要用 `<Outlet />` 来展示，故你需要在 A 组件中想要嵌套展示的部分添加上 `<Outlet />` 。

**Reply by xliez** [reply]
就是说我要在 /home 下加个父组件，根据路由路径是不是 /b 来展示 A组件 或 `<Outlet />`

**Reply by fz6m** [reply]
如果只是为了区分这两个路由的全部展示内容，而不需要嵌套路由的话，可以配置：

```ts
  routes: [
    { path: '/home', component: A },
    { path: '/home/b', component: B },
  ]
```

**Reply by xliez** [reply]
这样的话，自带的 ProLayout 菜单会变成2个，也没法直接和 PageContainer 的面包屑形成递进关系。

**Reply by fz6m** [reply]
如果需要搭配 max 的 layout 菜单自动生成，则需要将子路由配置放入 `routes` 形成嵌套路由格式，其中嵌套路由部分通过 `<Outlet />` 展示。
